src/CMakeLists.txt

```diff
@@ -842,3 +842,55 @@ if (IS_DIRECTORY "${PROJECT_SOURCE_DIR}/.git")
 endif()
 
 add_subdirectory(script)
+
+# add doxygen target
+find_package(Doxygen)
+if(DOXYGEN_FOUND)
+  set(DOXYGEN_FILE_PATTERNS *.cc *.c *.cpp *.C *.cxx *.c++ *.CC *.H *.h *.hh *.hpp)
+  set(DOXYGEN_SOURCE_BROWSER YES)
+  set(DOXYGEN_WARN_IF_UNDOCUMENTED NO)
+  set(DOXYGEN_CLANG_ASSISTED_PARSING YES)
+  set(DOXYGEN_CLANG_DATABASE_PATH .)
+  set(DOXYGEN_BUILTIN_STL_SUPPORT YES)
+  set(DOXYGEN_RECURSIVE YES)
+  set(DOXYGEN_QUIET YES)
+  set(DOXYGEN_GENERATE_LATEX NO)
+  doxygen_add_docs(doxygen
```

Comment:
might want to drop https://github.com/ceph/ceph/blob/master/Doxyfile? as this function generates a Doxyfile on the file in `CMAKE_CURRENT_BINARY_DIR` ?

Reply:
@tchaikov yes, will do in a separate PR. but probably going to adopt some parameters from there

Reply:
thanks. i always want to have less repeatings.

Reply:
done here: https://github.com/ceph/ceph/pull/37599